### PR TITLE
scripts/tag-release.sh: libuplink release tagging

### PR DESCRIPTION
**Pull request by jtolds**
a couple of arch review meetings ago we discussed how to make sure that it is much easier to get the release defaults for binaries, libraries, and so on. we already imperfectly solved the binary problem with the release.sh script, but (until now!) have not solved the problem of getting release defaults for people building from source.

the solution we seemed to all prefer was to make sure our tagged version commits check the release state into the source code.

this script aides in tagging commits with version tags and updating the source defaults. it still plays nicely with release.sh and our other build processes.

after this is merged we should configure github/go modules to prefer people use one of our tags instead of master (which will keep dev defaults). UPDATE: go modules does indeed already do the right thing.

Please describe the tests: no tests
 
Please describe the performance impact: no runtime performance impact

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: scripts/tag-release.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
```
Comment (egonelbre) [inline scripts/tag-release.sh]: Does this have to be a bash script :(

**Comment by egonelbre** [comment]
- [ ] Does the PR describe what changes are being made?
- [ ] Does the PR describe why the changes are being made?
- [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
- [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
- [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
- [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
- [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
- [ ] Does any documentation need updating?
- [ ] Do the database access patterns make sense?

**Comment by egonelbre** [comment]
Do we have documentation on our release process?

**Comment by jtolds** [comment]
I dunno, @littleskunk do we have anything?

I can add a comment about how this should be used in place of `git tag` somewhere I suppose

**Comment by littleskunk** [comment]
No we haven't documented it.

**Pull request merged**